test/Test.re

```diff
@@ -536,286 +698,278 @@ let core = [
+  });
+
+  test("The effect should not have been run again", ({expect}) => {
+    state := state^ |> update(Components.(<Div />)) |> executeSideEffects;
+
+    expect.int(effectCallCount^).toBe(1);
+  });
+
+  test(
+    "The effect dispose callback should have been called since the component was un-mounted.",
+    ({expect}) => {
+    expect.int(effectDisposeCallCount^).toBe(1)
+  });
+});
+
+describe("Test 'OnMount' effect in extra-nested component", ({test}) => {
+  let effectCallCount = ref(0);
+  let effectDisposeCallCount = ref(0);
+
+  let onEffect = () => incr(effectCallCount);
+  let onEffectDispose = () => incr(effectDisposeCallCount);
+
+  let state =
+    ref(
+      render(
+        Components.(
+          <Div>
             <Div>
               <EmptyComponentWithOnMountEffect onEffect onEffectDispose />
             </Div>
-          ),
-        )
-        |> executeSideEffects;
+          </Div>
+        ),
+      ),
+    );
 
-      expectInt(~label="The effect should've been run", 1, effectCallCount^);
+  /*
+   * When a parent-of-a-parent of a component with an OnMountEffect is removed,
+   * the OnMount effect doesn't get disposed on removal
+   */
+  test("The effect should've been run", ({expect}) => {
+    state := state^ |> executeSideEffects;
 
-      expectInt(
-        ~label="The dispose should not have been run yet",
-        0,
-        effectDisposeCallCount^,
-      );
+    expect.int(effectCallCount^).toBe(1);
+  });
 
-      testState
-      |> update(Components.(<Div />))
-      |> executeSideEffects
-      |> ignore;
-
-      expectInt(
-        ~label=
-          "The effect dispose callback should have been called since the component was un-mounted.",
-        1,
-        effectDisposeCallCount^,
-      );
-    },
-  ),
-  (
-    "Test 'OnMount' effect in extra-nested component",
-    `Quick,
-    () => {
-      let effectCallCount = ref(0);
-      let effectDisposeCallCount = ref(0);
-      let onEffect = () => incr(effectCallCount);
-      let onEffectDispose = () => incr(effectDisposeCallCount);
-
-      /*
-       * When a parent-of-a-parent of a component with an OnMountEffect is removed,
-       * the OnMount effect doesn't get disposed on removal
-       */
-      let testState =
-        render(
-          Components.(
-            <Div>
-              <Div>
-                <EmptyComponentWithOnMountEffect onEffect onEffectDispose />
-              </Div>
-            </Div>
-          ),
-        )
-        |> executeSideEffects;
+  test("The dispose should not have been run yet", ({expect}) => {
+    expect.int(effectDisposeCallCount^).toBe(0)
+  });
 
-      expectInt(~label="The effect should've been run", 1, effectCallCount^);
+  test("The effect should not have been run again", ({expect}) => {
+    state := state^ |> update(Components.(<Div />)) |> executeSideEffects;
 
-      expectInt(
-        ~label="The dispose should not have been run yet",
-        0,
-        effectDisposeCallCount^,
```

Comment:
Missing ?

Reply:
I think it's on line 953, the diff is a bit hard to read here.